docs/trace/extending-the-sdk/README.md

```diff
@@ -167,32 +166,40 @@ Writing an instrumentation library typically involves 3 steps.
       injecting/extracting `PropagationContext` explicitly again.)
 
 3. Third step is an optional step, and involves providing extension methods on
-   `TracerProviderBuilder`, to enable the instrumentation. This is optional, and
-   the below guidance must be followed:
-
-    1. If the instrumentation library requires state management tied to that of
-       `TracerProvider`, then it must register itself with the provider with the
-       `AddInstrumentation` method on the `TracerProviderBuilder`. This causes
-       the instrumentation to be created and disposed along with
-       `TracerProvider`. If the above is required, then it must provide an
-       extension method on `TracerProviderBuilder`. Inside this extension
-       method, it should call the `AddInstrumentation` method, and `AddSource`
-       method to enable its ActivitySource for the provider. An example
-       instrumentation using this approach is [SqlClient
+   `TracerProviderBuilder` and/or `IServiceCollection` to enable the
+   instrumentation. For help in choosing see: [Registration extension method
+   guidance for library
+   authors](#registration-extension-method-guidance-for-library-authors). This
+   is optional, and the below guidance should be followed:
+
+    * If the instrumentation library requires state management tied to that of
+       `TracerProvider` then it should:
+
+       * Implement `IDisposable`.
+
+       * Provide an extension method which calls `AddSource` (to enable its
+         `ActivitySource`) and `AddInstrumentation` (to enable state management)
+         on the `TracerProviderBuilder` being configured.
+
+       An example instrumentation using this approach is [SqlClient
        instrumentation](../../../src/OpenTelemetry.Instrumentation.SqlClient/TracerProviderBuilderExtensions.cs).
-       **CAUTION**: The instrumentation libraries requiring state management
-       are usually hard to auto-instrument. Therefore, they take the risk of not
-       being supported by [OpenTelemetry .NET Automatic Instrumentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation).
 
-    2. If the instrumentation library does not requires any state management
-       tied to that of `TracerProvider`, then providing `TracerProviderBuilder`
-       extension method is optional. If provided, then it must call `AddSource`
-       to enable its ActivitySource for the provider.
+       **CAUTION**: The instrumentation libraries requiring state management are
+       usually hard to auto-instrument. Therefore, they take the risk of not
+       being supported by [OpenTelemetry .NET Automatic
+       Instrumentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation).
+
+    * If the instrumentation library does not require any state management, then
```

Comment:
We could move this point and the one below under one bullet point. Something like this:

* If the instrumentation library does not require any state management, then providing an extension method is optional.
   - If an extension is provided it should call `AddSource` on the `TracerProviderBuilder` being configured to enable its `ActivitySource`.
   - If an extension is not provided, then the name of the `ActivitySource` used by the instrumented library must be documented so that end users can enable it by calling `AddSource` on the `TracerProviderBuilder` being configured. Changing the name of the source should be considered a breaking change.

Reply:
Updated